ci/ray_ci/doc/api.py

```diff
@@ -83,10 +86,28 @@ def from_autoclass(
         doc = doc.strip()
         if not doc.startswith(_SPHINX_AUTOCLASS_HEADER):
             return None
-        api_name = doc[len(_SPHINX_AUTOCLASS_HEADER) :].strip()
+        cls = doc[len(_SPHINX_AUTOCLASS_HEADER) :].strip()
+        api_name = f"{current_module}.{cls}" if current_module else cls
 
         return API(
-            name=f"{current_module}.{api_name}" if current_module else api_name,
+            name=API._fullname(api_name),
             annotation_type=AnnotationType.PUBLIC_API,
             code_type=CodeType.CLASS,
         )
+
+    @staticmethod
+    def _fullname(name: str) -> str:
+        """
+        Some APIs have aliases declared in __init__.py file (see ray/data/__init__.py
+        for example). This method converts the alias to full name. This is to make sure
+        out analysis can be performed on the same set of canonial names.
+        """
+        tokens = name.split(".")
+        attribute = importlib.import_module(tokens[0])
```

Comment:
this is just to import the first token, like as a package? I am confused how this works.

could you give an example (or a unit test) of how this function works?

Reply:
i added a test case in `test_api.py`; the high level idea is given the python path as string, e.g. a.b.c; this logic convert the string into a python object, by importing each token one by one ... import a, then import a.b, then import a.b.c, etc; once the python object is obtained, obtain its full name

Reply:
I am not sure if this is doing the right thing..

like:

````
>>> print(ray.init.__module__)
ray._private.worker
````

where the symbol is implemented does not always mean where the symbol is exposed / belongs to.

Reply:
maybe you can use this as a signature to determine that two symbols from different packages are the same one, but I am not sure that the alias is just a shortcut, or it is an exposed and even high-ranked interface.